skimage/restoration/deconvolution.py

```diff
@@ -375,16 +376,33 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
         convolve_method = fftconvolve
     else:
         convolve_method = convolve
-
     image = image.astype(np.float)
-    psf = psf.astype(np.float)
-    im_deconv = np.full(image.shape, 0.5)
+    psf = psf.astype(np.float)    
     psf_mirror = psf[::-1, ::-1]
-
-    for _ in range(iterations):
+    im_deconv = image
+    im_deconv_mid = np.zeros_like(image)
+    v = np.zeros_like(image)
+    #initial iteration
+    relative_blur = image / convolve_method(im_deconv, psf, 'same')
+    im_deconv_mid = im_deconv * convolve_method(relative_blur, psf_mirror,
+                                                    'same')
+    v = im_deconv_mid - im_deconv
+    np.clip(v, 0, v)           
+    im_deconv = im_deconv_mid
+    np.clip(im_deconv, 0, im_deconv)
+    
+    for iter in range(iterations-1):
```

Comment:
```suggestion
    for curr_iter in range(iterations):
```

(iter is a built-in function, it is best not to overwrite it)

Reply:
Now that `curr_iter` is no more used in the loop, you can hide it again by using `_` ;-)

Reply:
Yes, but various people don't like `_`, so I'm happy with either.

Reply:
Thank you all comments, @jni @rfezzani @hmaarrfk. I currently have no time for pytest (maybe also don't know how to do it, but I will have a try). If I have time in future (maybe next month), I will pr more kinds of deconvolution methods.